libs/shared/ebpf_shared_framework.h

```diff
@@ -26,6 +26,10 @@ CXPLAT_EXTERN_C_BEGIN
 #define EBPF_SYMBOLIC_DEVICE_NAME L"\\GLOBAL??\\EbpfIoDevice"
 #define EBPF_DEVICE_WIN32_NAME L"\\\\.\\EbpfIoDevice"
 
+#define ebpf_assert_ex(x) \
+    ebpf_assert(x);       \
+    __analysis_assume(x);
```

Comment:
Is it always safe to 'assume' instead of just using assert? Is there any risk of potential compiler assumptions if we are using analysis_assume?

Reply:
That is why I defined a new macro. Developers should use this sparingly, only when they want SAL to proceed.

Reply:
i suggest renaming this to ebpf_assert**_assume**, along the lines of NT_ASSERT_ASSUME, which is clearer than **_ex**.

Reply:
>Is there any risk of potential compiler assumptions if we are using analysis_assume?

The analysis assume should not influence the compiler, i.e. should not cause the type of misbehavior of `__assume` that Keith has been working to detangle.